src/gui/qgscollapsiblegroupbox.cpp

```diff
@@ -547,6 +549,9 @@ void QgsCollapsibleGroupBox::showEvent( QShowEvent * event )
 
 QString QgsCollapsibleGroupBox::saveKey() const
 {
+  if ( objectName().isEmpty() || ( mSettingGroup.isEmpty() && window()->objectName().isEmpty() ) )
+    return QString::null;  // cannot get a valid key
```

Comment:
That's why the qt docs recommend not using it - it's confusing.  QString() builds an empty but non null string, but a null QString is an empty string....!!